src/parlant/core/shots.py

```diff
@@ -0,0 +1,45 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import TypeVar, Generic, Sequence
+
+
+@dataclass
+class Shot:
+    description: str
+    """An explanation of what makes this shot interesting."""
+
+    @abstractmethod
+    def format(self) -> str: ...
```

Comment:
Not good to have format under the shot itself, we need to separate logic from presentation.